src/Utils/Authenticator.php

```diff
@@ -59,7 +59,14 @@ public function authenticate( stdClass $user ): WP_User {
 		}
 
 		if ( email_exists( $user->email ) ) {
-			return get_user_by( 'email', $user->email );
+			$user_wp = get_user_by( 'email', $user->email );
+
+			/**
+			 * Fires once the user has been authenticated.
```

Comment:
Add `@param` documentation for what arguments are passing to below action. Also, in which version it's been introduced. You can refer documentation of any action/filter in WP core codebase.